### PR TITLE
Detect solid-js via ANY element with "data-hk"

### PR DESCRIPTION
**Pull request by indeyets**
This approach is much more reliable and speed difference should be irrelevant as it happens one time on start and is still very fast.

Current solid-start (0.7.x, 1.0.x) doesn't hydrate "first child", and is usually applied only to the custom root.

p.s. also, I removed .js/.jsx extensions from eslint check, as eslint gave me error about lack of such files in repository.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         "----------------------TEST----------------------": "",
         "test:unit": "turbo run test --filter=./packages/*",
         "test:types": "turbo run test:types --filter=./packages/*",
-        "test:lint": "eslint --ignore-path .gitignore --max-warnings 0 packages/**/*.{js,ts,tsx,jsx}",
+        "test:lint": "eslint --ignore-path .gitignore --max-warnings 0 packages/**/*.{ts,tsx}",
```
Comment (thetarnav) [inline package.json:20]: Whats exactly the error? I haven't seen one like that myself.
Also there are some `.js` files and there will probably be more.

Comment (indeyets) [reply package.json:20]: solid-devtools 2024-04-01 12-36-59

File: packages/shared/src/detect.ts [hunk 2]
```diff
@@ -19,8 +19,8 @@ export async function detectSolid(): Promise<boolean> {
     if ($hy && typeof $hy === 'object' && 'completed' in $hy && $hy.completed instanceof WeakSet)
         return true
 
-    const bodyFirstEl = document.body.firstElementChild
-    if (bodyFirstEl && bodyFirstEl.hasAttribute(DATA_HYDRATION_KEY)) return true
+    if (document.querySelectorAll('[' + DATA_HYDRATION_KEY + ']').length > 0)
+        return true
```
Comment (thetarnav) [inline packages/shared/src/detect.ts:23]: Wouldn't `document.querySelector` be better here?
I only need to know if there are ANY elements like that, I don't need to search for ALL.

Comment (indeyets) [reply packages/shared/src/detect.ts:23]: sure, I can redo this 👍

File: package.json (hunk 1 above)
Comment (thetarnav) [reply package.json:20]: Can you try surrounding the original pattern with `\"`?
This has to be some os issue, because I don't have that behavior.

Comment (indeyets) [reply package.json:20]: yup. it does the trick. good catch.
should I wrap all other globs in quotes as well?

Comment (thetarnav) [reply package.json:20]: Is is necessary anywhere else? It might be eslint specific.

**Pull request merged**